project/src/components/Map/Map.tsx

```diff
@@ -1,11 +1,57 @@
-import React from 'react';
+import { Icon, Marker } from 'leaflet';
+import { useRef, useEffect } from 'react';
+import { iconAnchor, IconSize, MapIconUrl } from '../../const';
+import { useMap } from '../../hooks';
+import { City, Points } from '../../types/map';
+import 'leaflet/dist/leaflet.css';
 
 type MapProps = {
   className: 'cities' | 'property',
+  city: City,
+  points: Points,
+  activeCard: number | null,
 };
 
-function Map({ className }: MapProps) {
-  return <section className={`${className}__map map`} />;
+const defaultCustomIcon = new Icon({
+  iconUrl: MapIconUrl.Default,
+  iconSize: [IconSize.width, IconSize.height],
+  iconAnchor: [iconAnchor.width / 2, iconAnchor.height],
+});
+
+const currentCustomIcon = new Icon({
+  iconUrl: MapIconUrl.Active,
+  iconSize: [IconSize.width, IconSize.height],
+  iconAnchor: [iconAnchor.width / 2, iconAnchor.height],
+});
+
+function Map({ className, city, points, activeCard }: MapProps) {
+  const mapRef = useRef(null);
+  const map = useMap(mapRef, city);
+
+  useEffect(() => {
+    if (map && activeCard !== null) {
```

Comment:
От активного оффера рендер меток не зависит они будут просто все синие
```suggestion
    if (map) {
```